### ISSUE TITLE
Release 1.0.50 not installing or updating

### ISSUE THREAD

**Issue by jp-lesage** [brandonp2412/Flexify#14]
Hi!

I downloaded the 1.0.50 release (the latest at the time of creating this ticket), and it errors out when installing or upgrading the installed version.

I downloaded the package multiple times and got the same error each time, and it complains about the package being invalid. I am using the `app-arm64-v8a-release.apk` file, but I didn't try the others (since I don't have devices for those). A reboot of the device and a cleaning of the cache did not fix the issue.

Sixo online APK analyzer can parse it just fine (https://sisik.eu/apk-tool), but my device doesn't agree with it... I'm running a Samsung S24 Ultra with Android 14 and the latest patch installed from Samsung.

That's the message I get:
![SmartSelect_20240419_140006_Package installer.png](https://github.com/brandonp2412/Flexify/assets/53324016/2c4f1079-74ff-4404-b210-19dc64fc94fe)

I couldn't find any logs anywhere (either app logs or system logs) where I could see what's the actual error.

Thanks!

**Comment by jp-lesage** [comment]
One thing I forgot to mention is that I'm updating from 1.0.44.

I just tried reinstalling 1.0.44, to make sure it works, and it does. Every subsequent release, starting from (and including) 1.0.45, fails to install or upgrade with the same error.

Thanks!

**Comment by brandonp2412** [comment]
Do the other apks fail as well? Maybe I've tagged them wrong

**Comment by brandonp2412** [comment]
Hmm doubtful they are tagged wrong since the arm64 installs on my pixel fine. I do notice a lot of this error in online forums mentioning your phone version. I'll add the universal APK to the next release and we can see if that works.

**Comment by brandonp2412** [comment]
Does the universal release here https://github.com/brandonp2412/Flexify/releases/download/1.0.51/app-release.apk work?

**Comment by jp-lesage** [comment]
Just tried the universal release and... same error. I'm getting suspicious there is something else at play. If it's a setting that changed itself during an update, I'll be really upset.

I'll do some tests with other APKs and see if they do the same thing.

Thanks!

**Comment by brandonp2412** [comment]
Not sure if it's relevant to your device but someone I know managed to install it on a samsung S22. Weird that you aren't seeing any relevant logs in ADB when you install it, but I'm pretty sure I've run into similar problems on my previous app Massive. However that issue was to do with the signatures changing and basically I couldn't see any info about it in adb. Also that one worked once i completely uninstalled + reinstalled so this bug is indeed odd.

**Comment by jp-lesage** [comment]
While waiting for my McDonald's order, I fiddled around with a few things and ended up uninstalling and reinstalling the app a few times. Installing the app started to work consistently after I cleaned the cache a few times, which is odd.

It seems that something, perhaps a signature change as you mentioned, caused anything below v1.0.45 to not be updatable with the newer releases. I'm surprised that no logs were generated by the system when that happens, and it just "fails" with no details.

I tried both the arm64 and the universal installer, and both work (as long as the app is fully uninstalled before when the version is 1.0.44 or less). I'll check the update on the next release, but I'm confident it'll work. If not, nuke and reinstall will be the first thing I do.

I think we can mark this issue as closed :)

Thanks!
